toy_settings/django_back_end/unit_of_work.py

```diff
@@ -19,10 +20,18 @@ def atomic(self) -> Iterator[None]:
 
     def handle(self, event: events.Event) -> None:
         event_type, event_type_version = models.EVENT_TYPES[type(event)]
-        models.Event.objects.create(
+        new_event = models.Event.objects.create(
             event_type=event_type,
             event_type_version=event_type_version,
             timestamp=event.timestamp,
             key=event.key,
             payload=models.Event.payload_converter.dumps(event),
         )
+        try:
+            models.Sequence.objects.create(
+                event=new_event,
+                key=event.key,
+                idx=event.idx,
+            )
+        except IntegrityError as exc:
```

Comment:
Future improvement: It might be worth distinguishing between the error we'd get from a constraint error, and the one that would come from a string longer than 100 chars.

Reply:
Yeah, I vaguely remember you had some code that could do that neatly, but I can't find it anymore